File: Ex2Num8.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n int // количество цветов
	fmt.Print("Введите количество цветов: ")
	fmt.Scan(&n)

	water := make([]int, n) // потребность каждого цветка
	fmt.Println("Введите потребность каждого цветка через пробел:")
	for i := 0; i < n; i++ {
		fmt.Scan(&water[i])
	}

	var capacity int // объем лейки
	fmt.Print("Введите объем лейки: ")
	fmt.Scan(&capacity)

	steps := 0
	currentWater := capacity
	position := -1 // начинаем с реки

	for i := 0; i < n; i++ {
		if currentWater >= water[i] {
			steps += int(math.Abs(float64(position - i)))
			currentWater -= water[i]
			position = i
		} else {
			// идем к реке и обратно
			steps += int(math.Abs(float64(position - (-1)))) // до реки
			position = -1
			currentWater = capacity

			steps += int(math.Abs(float64(position - i))) // обратно к цветку
			currentWater -= water[i]
			position = i
		}
	}

	fmt.Printf("Общее количество шагов: %d\n", steps)
}

File: Ex2Num8.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int main() {
    int n; // количество цветов
    cout << "Введите количество цветов: ";
    cin >> n;

    vector<int> water(n); // потребность каждого цветка
    cout << "Введите потребность каждого цветка через пробел:\n";
    for (int i = 0; i < n; ++i) {
        cin >> water[i];
    }

    int capacity; // объем лейки
    cout << "Введите объем лейки: ";
    cin >> capacity;

    int steps = 0;
    int current_water = capacity;
    int position = -1; // начинаем с реки

    for (int i = 0; i < n; ++i) {
        if (current_water >= water[i]) {
            steps += abs(position - i);
            current_water -= water[i];
            position = i;
        } else {
            // идем к реке и обратно
            steps += abs(position - (-1)); // до реки
            position = -1;
            current_water = capacity;

            steps += abs(position - i); // обратно к цветку
            current_water -= water[i];
            position = i;
        }
    }

    cout << "Общее количество шагов: " << steps << endl;
    return 0;
}
